File: app/src/api/product-catalog/pad-series/data-transfer-objects/pad.ts
import { Brand, PadCategory, Stars } from '@/api';
import { PadSeries } from '@/api/product-catalog/pad-series/data-transfer-objects/pad-series';
import { Image } from '@/api/product-catalog/common/data-transfer-objects/image';
import { PadFinish } from './pad-finish';
import { PadCut } from './pad-cut';
import { PadSize } from './pad-size';
import { PadMaterial } from './pad-material';
import { PadTexture } from './pad-texture';
import { PolisherType } from './polisher-type';
import { Rating } from './rating';

export class Pad {
    get label() {
        return `${this.series.brand.name} ${this.series.name} ${this.name}`
    }

    constructor(
        public id: string,
        public series: PadSeries,
        public name: string,
        public category: PadCategory,
        public cut: PadCut | null,
        public finish: PadFinish | null,
        public material: PadMaterial,
        public texture: PadTexture,
        public sizes: PadSize[],
        public polisherTypes: PolisherType[],
        public rating: Rating,
        public image: Image | null = null,
    ) { }
}

File: api/src/Domain/ProductCatalog/Pads/Entities/Pad.cs
using System;
using System.Collections.Generic;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class Pad : Entity<Pad> {
        public string Name { get; set; }
        public PadCategory Category { get; set; }
        public PadMaterial Material { get; set; }
        public PadTexture Texture { get; set; }
        public List<PolisherType> PolisherTypes { get; set; }
        public DataUrlImage? Image { get; set; }

        public Pad(string name, PadCategory category, PadMaterial material, PadTexture texture, List<PolisherType>? polisherTypes = null, DataUrlImage? image = null) {
            Id = Guid.NewGuid();
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            PolisherTypes = polisherTypes ?? new List<PolisherType>();
            Image = image;
        }

        public Pad(Guid id, string name, PadCategory category, PadMaterial material, PadTexture texture, List<PolisherType>? polisherTypes = null, DataUrlImage? image = null) {
            Id = id;
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            PolisherTypes = new List<PolisherType>();
            Image = image;
        }
    }
}